spinta/manifests/open_api/helpers.py

```diff
@@ -2,99 +2,325 @@
     """
-    return re.sub(r'{([^{}]+)}', lambda match: f'{{{to_code_name(match.group(1))}}}', endpoint)
+    return re.sub(r"{([^{}]+)}", lambda match: f"{{{to_code_name(match.group(1))}}}", endpoint)
 
 
 def read_file_data_and_transform_to_json(path: Path) -> dict:
     with Path(path).open() as file:
         return json.load(file)
 
 
+def resolve_refs(schema: dict, root: dict) -> dict:
+    if isinstance(schema, dict):
+        if "$ref" in schema:
+            ref_path = schema["$ref"].lstrip("#/").split("/")
+            resolved = root
+            for part in ref_path:
+                resolved = resolved[part]
+            return resolve_refs(resolved, root)
+
+        return {key: resolve_refs(value, root) for key, value in schema.items()}
+
+    return schema
+
+
 def get_namespace_schema(info: dict, title: str, dataset_prefix: str) -> Generator[tuple[None, dict], None, None]:
-    yield None, {
-        'type': 'ns',
-        'name': dataset_prefix,
-        'title': info.get('summary', title),
-        'description': info.get('description', ''),
-    }
+    yield (
+        None,
+        {
+            "type": "ns",
+            "name": dataset_prefix,
+            "title": info.get("summary", title),
+            "description": info.get("description", ""),
+        },
+    )
 
 
 def get_resource_parameters(parameters: list[dict]) -> dict[str, dict]:
     resource_parameters = {}
     for index, value in enumerate(parameters):
-        name = value['name']
-        location = value['in'] if value['in'] in SUPPORTED_PARAMETER_LOCATIONS else ''
-        resource_parameters[f'parameter_{index}'] = {
-            'name': to_code_name(name),
-            'source': [name],
-            'prepare': [Expr(location)],
-            'type': 'param',
-            'description': value.get('description', ''),
+        name = value["name"]
+        location = value["in"] if value["in"] in SUPPORTED_PARAMETER_LOCATIONS else ""
+        resource_parameters[f"parameter_{index}"] = {
+            "name": to_code_name(name),
+            "source": [name],
+            "prepare": [Expr(location)],
+            "type": "param",
+            "description": value.get("description", ""),
         }
 
     return resource_parameters
 
 
-def get_dataset_schemas(data: dict, dataset_prefix: str) -> Generator[tuple[None, dict]]:
+model_deduplicator = Deduplicator()
+
+class Model:
+    def __init__(
+        self,
+        dataset: str,
+        resource: str,
+        basename: str,
+        json_schema: dict,
+        source: str | None = None,
+        parent: Model | None = None
+    ) -> None:
+        self.dataset: str = dataset
+        self.resource: str = resource
+        self.basename: str = basename
+        self.json_schema: dict = json_schema
+        self.source: str = source if source is not None else basename
+        self.title: str = self.json_schema.get("title")
+        self.description: str = self.json_schema.get("description")
```

Comment:
Can `title`/`description` be `None`? Than the typing would be innacurrate

Might want this:
```suggestion
        self.title: str = self.json_schema.get("title", "")
        self.description: str = self.json_schema.get("description", "")
```